core/src/main/java/dk/alexandra/fresco/logging/PerformanceLogger.java

```diff
@@ -12,24 +12,26 @@
  *
  */
 public interface PerformanceLogger {
-
-  public Logger log = LoggerFactory.getLogger(PerformanceLogger.class);
-  public DecimalFormat df = new DecimalFormat("#.00");
   
-  public enum Flag {
-    LOG_NETWORK, LOG_RUNTIME, LOG_NATIVE_BATCH;
-
-    public static final EnumSet<Flag> ALL_OPTS = EnumSet.allOf(Flag.class);
-  }
-
   /**
-   * Prints any performance numbers picked up.
+   * Resets any counters/maps/lists used. 
    */
-  public abstract void printPerformanceLog(int myId);
+  public abstract void reset();    
   
   /**
-   * Resets any counters/maps/lists used. 
+   * Produce a key-value map of logged values. The content
+   * of the returned map is context dependent. 
    */
-  public abstract void reset();    
+  public abstract Map<String, Object> getLoggedValues(int myId);
```

Comment:
Given PerformanceLogger counts things the map could even be to integer?

Reply:
The NetworkLogger currently also logs bytes received pr. party, so the map should flexible enough to handle other datatypes, if we wish to support that level of logging.

Reply:
Just create a composite key with "BytesRecieved-<id>"